### PR TITLE
Replace jvm-example by two plugin examples

### PR DESCRIPTION
**Pull request by tlrx**
This pull request replaces the `jvm-example` plugin (from the _jvm/site_ plugins era) by two new plugins: a `custom-settings` that shows how to register and use custom settings (including secured settings) in a plugin, and `rest-handler` plugin that shows how to register a rest handler.

The two plugins now reside in the `plugins/examples` project. They can serve as sample plugins for users, a special attention has been put on documentation. The packaging tests have been adapted to use the `custom-settings` plugin.

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestClient.java [hunk 1]
```diff
@@ -100,8 +100,8 @@ public ClientYamlTestResponse callApi(String apiName, Map<String, String> params
                 queryStringParams.put(entry.getKey(), entry.getValue());
                 apiRequiredParameters.remove(entry.getKey());
             } else {
-                throw new IllegalArgumentException(
-                        "path/param [" + entry.getKey() + "] not supported by [" + restApi.getName() + "] " + "api");
+                //throw new IllegalArgumentException(
```
Comment (nik9000) [inline test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestClient.java]: Leftover?

**Review by nik9000 — COMMENTED** [review]
I like it! I left a single comment, but that comment is probably important.

File: test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestClient.java (hunk 1 above)
Comment (tlrx) [reply test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestClient.java]: Definitely, thanks!

**Review by rjernst — APPROVED** [review]
Thank you @tlrx! This is great!

**Comment by tlrx** [comment]
Thanks @nik9000 and @rjernst

**Pull request merged**